bigtable/client/filters.h

```diff
@@ -214,26 +214,158 @@ class Filter {
+    return tmp;
+  }
+
+  //@{
+  /**
+   * @name Less common range filters.
+   *
+   * Cloud Bigtable range filters can include or exclude the limits of the
+   * range.  In most cases applications use [start,end) ranges, and the
+   * ValueRange() and ColumnRange() functions are offered to support the
+   * common case.  For the less common cases where the application needs
+   * different ranges the following functions are available.
+   */
+  /**
+   * Return a filter that accepts values in the [@p start, @p end) range.
+   *
+   * TODO(#84) - document what happens if end < start
+   */
+  static Filter ValueRangeLeftOpen(std::string start, std::string end) {
+    Filter tmp;
+    auto& range = *tmp.filter_.mutable_value_range_filter();
+    range.set_start_value_open(std::move(start));
+    range.set_end_value_closed(std::move(end));
+    return tmp;
+  }
+
+  /**
+   * Return a filter that accepts values in the [@p start, @p end] range.
+   *
+   * TODO(#84) - document what happens if end < start
+   */
+  static Filter ValueRangeRightOpen(std::string start, std::string end) {
+    Filter tmp;
+    auto& range = *tmp.filter_.mutable_value_range_filter();
+    range.set_start_value_closed(std::move(start));
+    range.set_end_value_open(std::move(end));
+    return tmp;
+  }
+
+  /**
+   * Return a filter that accepts values in the [@p start, @p end] range.
+   *
+   * TODO(#84) - document what happens if end < start
+   */
+  static Filter ValueRangeClosed(std::string start, std::string end) {
+    Filter tmp;
+    auto& range = *tmp.filter_.mutable_value_range_filter();
+    range.set_start_value_closed(std::move(start));
+    range.set_end_value_closed(std::move(end));
+    return tmp;
+  }
+
+  /**
+   * Return a filter that accepts values in the (@p start, @p end) range.
+   *
+   * TODO(#84) - document what happens if end < start
+   */
+  static Filter ValueRangeOpen(std::string start, std::string end) {
+    Filter tmp;
+    auto& range = *tmp.filter_.mutable_value_range_filter();
+    range.set_start_value_open(std::move(start));
+    range.set_end_value_open(std::move(end));
+    return tmp;
+  }
+
+  /**
+   * Return a filter that accepts columns in the [@p start, @p end) range
+   * within the @p column_family.
+   *
+   * TODO(#84) - document what happens if end < start
+   */
+  static Filter ColumnRangeRightOpen(std::string column_family,
+                                     std::string start, std::string end) {
+    Filter tmp;
+    auto& range = *tmp.filter_.mutable_column_range_filter();
+    range.set_family_name(std::move(column_family));
+    range.set_start_qualifier_closed(std::move(start));
+    range.set_end_qualifier_open(std::move(end));
+    return tmp;
+  }
+
+  /**
+   * Return a filter that accepts columns in the (@p start, @p end] range
+   * within the @p column_family.
+   *
+   * TODO(#84) - document what happens if end < start
+   */
+  static Filter ColumnRangeLeftOpen(std::string column_family,
```

Comment:
@sduskis, @garye, @DanielMahu — do we actually support column range queries? I thought we only support row range queries, and column families and column qualifiers need to be specified explicitly, one at a time.

Reply:
ColumnRangeFilters can be applied just like any other filter:
https://github.com/googleapis/googleapis/blob/master/google/bigtable/v2/data.proto#L390

Reply:
OK, got it, so column families are discrete (must be specified explicitly), while column qualifiers are continuous (can be selected individually, or via a range).